src/VoiceConnection.ts

```diff
@@ -348,6 +361,34 @@ export class VoiceConnection extends EventEmitter {
 		if (this.state.status !== VoiceConnectionStatus.Ready) return false;
 		this.state.networking.setSpeaking(enabled);
 	}
+
+	/**
+	 * Subscribes to an audio player, allowing the player to play audio on this voice connection.
+	 *
+	 * @param player The audio player to subscribe to
+	 * @returns The created subscription
+	 */
+	public subscribe(player: AudioPlayer) {
+		if (this.state.status === VoiceConnectionStatus.Destroyed) return;
+
+		const unsubscribe = () => {
+			if (this.state.status !== VoiceConnectionStatus.Destroyed && this.state.subscription === subscription) {
```

Comment:
The fact this depends on `subscription`, which is defined a few lines below, is confusing and could lead to potential issues in the hypothetical case that the callback were to be called before `player.subscribe` returns.

However, other than making the callback take one argument, or removing this condition altogether, I don't know a better alternative. cc: @iCrawl